Sources/StreamChat/Query/ChannelListQuery.swift

```diff
@@ -58,80 +61,127 @@ public enum InviteFilterValue: String, FilterValue {
+        ($0 as? ChannelPayload)?.channel.name
+    }}
 
     /// A filter key for matching the `image` value.
-    static var imageURL: FilterKey<Scope, URL> { "image" }
+    static var imageURL: FilterKey<Scope, URL> { .init(rawValue: "image") {
+        ($0 as? ChannelPayload)?.channel.imageURL
+    }}
 
     /// A filter key for matching the `type` value.
     /// Supported operators: `in`, `equal`
-    static var type: FilterKey<Scope, ChannelType> { "type" }
+    static var type: FilterKey<Scope, ChannelType> { .init(rawValue: "type") {
+        if let typeRawValue = ($0 as? ChannelPayload)?.channel.typeRawValue {
+            return ChannelType(rawValue: typeRawValue)
+        } else {
+            return nil
+        }
+    }}
 
     /// A filter key for matching the `lastMessageAt` value.
     /// Supported operators: `equal`, `greaterThan`, `lessThan`, `greaterOrEqual`, `lessOrEqual`, `notEqual`
-    static var lastMessageAt: FilterKey<Scope, Date> { "last_message_at" }
+    static var lastMessageAt: FilterKey<Scope, Date> { .init(rawValue: "last_message_at") {
+        ($0 as? ChannelPayload)?.channel.lastMessageAt
+    }}
 
     /// A filter key for matching the `createdBy` value.
     /// Supported operators: `equal`
-    static var createdBy: FilterKey<Scope, UserId> { "created_by_id" }
+    static var createdBy: FilterKey<Scope, UserId> { .init(rawValue: "created_by_id") {
+        ($0 as? ChannelPayload)?.channel.createdBy?.id
+    }}
 
     /// A filter key for matching the `createdAt` value.
     /// Supported operators: `equal`, `greaterThan`, `lessThan`, `greaterOrEqual`, `lessOrEqual`, `notEqual`
-    static var createdAt: FilterKey<Scope, Date> { "created_at" }
+    static var createdAt: FilterKey<Scope, Date> { .init(rawValue: "created_at") {
+        ($0 as? ChannelPayload)?.channel.createdAt
+    }}
 
     /// A filter key for matching the `updatedAt` value.
     /// Supported operators: `equal`, `greaterThan`, `lessThan`, `greaterOrEqual`, `lessOrEqual`, `notEqual`
-    static var updatedAt: FilterKey<Scope, Date> { "updated_at" }
+    static var updatedAt: FilterKey<Scope, Date> { .init(rawValue: "updated_at") {
+        ($0 as? ChannelPayload)?.channel.updatedAt
+    }}
 
     /// A filter key for matching the `deletedAt` value.
     /// Supported operators: `equal`, `greaterThan`, `lessThan`, `greaterOrEqual`, `lessOrEqual`, `notEqual`
-    static var deletedAt: FilterKey<Scope, Date> { "deleted_at" }
+    static var deletedAt: FilterKey<Scope, Date> { .init(rawValue: "deleted_at") {
+        ($0 as? ChannelPayload)?.channel.deletedAt
+    }}
 
     /// A filter key for querying hidden channels.
     /// Supported operators: `equal`
-    static var hidden: FilterKey<Scope, Bool> { "hidden" }
+    // TODO: should it be using the ChannelPayload.isHidden or ChannelPayload.channel.isHidden
+    static var hidden: FilterKey<Scope, Bool> { .init(rawValue: "hidden") {
+        ($0 as? ChannelPayload)?.isHidden
```

Comment:
Is this the right keypath for the members or shall we instead get from the `($0 as? ChannelPayload)?.channel.isHidden`?